src/content_cao.cpp

```diff
@@ -1786,6 +1831,14 @@ void GenericCAO::processMessage(const std::string &data)
 	}
 }
 
+void GenericCAO::knockback(const v3f &direction, f32 time_knockback)
+{
+	m_knockback_timer = time_knockback;
```

Comment:
in fact it's a little bit more complicated, the client do the animation on all clients simultaneously and is unlocked at the end of the timer, unlocking sending of TOSERVER_MOVE packets which interfere during this knockback